### PR TITLE
Add Corda Configuration section

### PR DESCRIPTION
**Pull request by nadinequinr3**
Part of a larger piece of work to document Corda configuration: https://r3-cev.atlassian.net/browse/DOC-4865

Next steps for Beta 2:
- Add pages for remaining sections
- Review the descriptions pulled from the schemas in https://github.com/corda/corda-api/tree/release/os/5.0-Beta2/data/config-schema/src/main/resources/net/corda/schema/configuration and raise PRs for Development
- Review the documentation that populates the config endpoint - https://docs.r3.com/en/platform/corda/5.0-beta/rest-api/C5_OpenAPI.html#tag/Configuration-API - and raise PRs for Development

Most likely post Beta 2:
- Write Hugo shortcodes to generate examples for each section.
- Write a tutorial page to retrieve the current configuration 
- Write a tutorial page to update the configuration 

Preview currently available here: https://nadine.preview.docs.r3.com/en/platform/corda/5.0-beta/operating/overview.html

FYI @driessamyn @mattbradburyr3

### CODE REVIEW

**Review by Niamh25 — APPROVED** [review]

File: layouts/shortcodes/generate-ref-docs.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          {{ end }}
+          {{ with $value.allOf }}
+               {{ partial "conditional-field-desc-ref-docs" (dict "value" $value "key" $key "url" $.Params.url) }}
+          {{ end }}
+          </ul>      
+       </ul>
+       {{ end }}
+       {{ end }}
+       </ul>      
+{{ end }}
```
Comment (yarivus) [inline layouts/shortcodes/generate-ref-docs.html:22]: Please indent 4 spaces where needed ... some have less ... and other lines have missing indent 
(For increased readability)

Comment (yarivus) [reply layouts/shortcodes/generate-ref-docs.html:22]:
```suggestion
{{ with $.Params.url }}
    {{ $json := getJSON $.Params.url }}
    <p>{{ $json.description }}</p> 
    {{ range $key, $value := $json.properties }}
        {{ with $value }}
            <ul>
                {{ partial "field-desc-ref-docs" (dict "value" $value "key" $key "url" $.Params.url) }}
                <ul>
                {{ range $childk, $childv := $value.properties }}
                    {{ with $childv }}
                        {{ partial "field-desc-ref-docs" (dict "value" $childv "key" $childk) }}
                    {{ end }}
                {{ end }}
                {{ with $value.allOf }}
                    {{ partial "conditional-field-desc-ref-docs" (dict "value" $value "key" $key "url" $.Params.url) }}
                {{ end }}
                </ul>      
           </ul>
       {{ end }}
   {{ end }}
   </ul>      
{{ end }}
```

Comment (yarivus) [reply layouts/shortcodes/generate-ref-docs.html:22]: Take note that there's an additional `</ul>`

File: layouts/partials/conditional-field-desc-ref-docs.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                                        {{ with $childv }}
+                                            {{ partial "field-desc-ref-docs" (dict "value" $childv "key" $childk) }}
+                                        {{ end }}
+                                    {{ end }}
+                                </ul>
+                            {{ end }}                        
+                        </ul>
+                        {{ end }}
+                    {{ end }}
+
```
Comment (yarivus) [inline layouts/partials/conditional-field-desc-ref-docs.html:43]: remove redundant newlines

File: layouts/partials/referenced-field-desc-ref-docs.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{{ $refurl := .url }}
+{{ $refjson := getJSON $refurl }}
+<ul>
+{{ range $key, $value := $refjson.properties }}
+    {{ with $value }}
+        <li><code>{{ $key }}</code> - {{ $value.description}}
```
Comment (yarivus) [inline layouts/partials/referenced-field-desc-ref-docs.html:6]:
```suggestion
        <li>
            <code>{{ $key }}</code> - {{ $value.description}} 
```

File: layouts/partials/referenced-field-desc-ref-docs.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+{{ range $key, $value := $refjson.properties }}
+    {{ with $value }}
+        <li><code>{{ $key }}</code> - {{ $value.description}} 
+            {{ with $value.enum }}
+                This must be set to one of the following values:
+                <ul>
+                {{ range $enumval := $value.enum }}
+                    <li>
+                    {{ $enumval }}
+                    </li>
```
Comment (yarivus) [inline layouts/partials/referenced-field-desc-ref-docs.html:13]:
```suggestion
                    <li> {{ $enumval }} </li>
```
For short ones ... it's easier to read a 1-line ;)

Comment (yarivus) [reply layouts/partials/referenced-field-desc-ref-docs.html:13]: Fix other occurrences

**Review by yarivus — CHANGES_REQUESTED** [review]

File: layouts/shortcodes/generate-ref-docs.html (hunk 1 above)
Comment (nadinequinr3) [reply layouts/shortcodes/generate-ref-docs.html]: Thanks. Think I was aiming for 3, which is habit from markdown.

**Review by yarivus — APPROVED** [review]
LGTM

**Review by Niamh25 — APPROVED** [review]

**Pull request merged**